src/lib.rs

```diff
@@ -1,65 +1,166 @@
+/// # Fantom libtransport
+///
+/// This library defines a trait for the management and definition of specific transport prototcols
+/// (such as TCP/IP or UDP). This repository solely defines the trait and provides no further
+/// functionality. If you want a specific implementation, please check the transport-tcp repo for a
+/// complete implementation: https://github.com/Fantom-foundation/libtransport-tcp.
+///
+/// Currently, two traits are defined: TransportConfiguration and Transport.
+///
+/// # TransportConfiguration
+///
+/// Currently the TransportConfiguration trait requires the definition of a type: Data. Data can
+/// be of any type which needs to be transmitted across the network. An example of a 'Data'
+/// definition can be:
+///
+/// ```
+/// // An arbritrary data type which should be transmitted across the network
+/// pub struct Data(pub u32);
+///
+/// ```
+/// This trait also comes with two functions which need to be implemented: 'new' and
+/// 'set_bind_net_addr'. These will be addressed below.
+///
+/// A transport configuration is essential for the Transport trait to function properly.
+///
+/// # Transport
+///
+/// The Transport trait defines the functionality for connecting with other devices in the same
+/// network. This trait will handle the barebones methods of:
+///
+/// a) Creating a set of peers
+/// b) Sending data between multiple peers
+/// c) Broadcasting messages to all peers within the same network.
+///
+/// The Transport trait requires the definition of a number of types to work properly:
+///
+/// Id: The unqiue ID type of the peer in question. This can be as simple as:
+/// ```
+/// // Another arbitrary type for holding ID data
+/// pub struct Id(pub u32);
+/// ```
+/// <b>Data:</b> The data being transmitted between peers - same as the data struct defined above.
+/// <b>Error:</b> An error type that can be returned by implementations of the 'PeerList' trait
+/// <b>Pl:</b> A struct which implementes the PeerList trait (defined in the libcommon repo:
+/// https://github.com/Fantom-foundation/libcommon-rs )
+/// <b>Configuration:</b> The TransportConfiguration type required to make the function work.
+///
+/// Finally, the Transport trait implements three methods: 'new', 'send', and 'broadcast'
```

Comment:
and requires implementation of `Stream` trait from async/.await framework of the Rust language (nightly).